PopcornEditor/src/butter.js

```diff
@@ -922,67 +922,127 @@ window.Butter = {
-                  track.trackEvents.push({
-                    id: "TrackEvent1",
-                    type: "sequencer",
-                    popcornOptions: {
-                      source: [ data.source ],
-                      start: 0,
-                      end: data.duration,
-                      title: data.title,
-                      from: data.from,
-                      duration: data.duration,
-                      target: "video-container",
-                      fallback: "",
-                      zindex: 1000,
-                      id: "TrackEvent1",
-                      linkback: data.linkback,
-                      thumbnailSrc: data.thumbnail
-                    }
-                  });
+            var media = savedData.media[ 0 ];
+            if (media)
+              media.duration = 0;
+
+            var num2get = initialMediaList.length;
+            $(initialMediaList).each(function(idx,initialMediaSource){
+              // If we successfully retrieve data for that initial media we will hand write it
+              // into the default project data before the import.
+              // xxxx 2+ media is race condition adding now!
+
+              MediaUtil.getMetaData( initialMediaSource, function onSuccess( data ) {
+                var media = savedData.media[ 0 ],
+                    track;
+
+                if ( media ) {
+                  media.url = "#t=," + data.duration;
+                  media.duration += data.duration;
+
+                  track = media.tracks[ 0 ];
+
+                  if ( track ) {
+                    track.trackEvents.push({
+                      id: "TrackEvent"+idx,
+                      type: "sequencer",
+                      popcornOptions: {
+                        source: [ data.source ],
+                        start: trackTime,
+                        end: trackTime + data.duration,
+                        title: data.title,
+                        from: data.from,
+                        duration: data.duration,
+                        target: "video-container",
+                        fallback: "",
+                        zindex: 1000,
+                        id: "TrackEvent"+idx,
+                        linkback: data.linkback,
+                        thumbnailSrc: data.thumbnail
+                      }
+                    });
+                    trackTime += data.duration;
 
-                  media.clipData = media.clipData || {};
+                    media.clipData = media.clipData || {};
 
-                  // Don't forget to add the clip data!
-                  if ( !media.clipData[ data.source ] ) {
-                    media.clipData[ data.source ] = data;
+                    // Don't forget to add the clip data!
+                    if ( !media.clipData[ data.source ] ) {
+                      media.clipData[ data.source ] = data;
+                    }
                   }
+                }
 
+                if ( ! --num2get ) {
```

Comment:
sorry, that was me trying to be (too) cute.  i was hesitant to commit but wanted 2b kewl
if ( (--num2get) == 0) {